crates/swc/tests/tsc-references/recursiveInitializer.2.minified.js

```diff
@@ -1,2 +1,2 @@
 //// [recursiveInitializer.ts]
-var n1 = n1++, n2 = n2 + n2, n3 = n3 + n3, s2 = s2 + s2, s3 = s3 + s3, b3 = !b3 || b3, b4 = !b4 && b4;
+var n1 = n1++, n2 = n2 + n2, n3 = n3 + n3, s1 = s1 + "", s2 = s2 + s2, s3 = s3 + s3, s4 = "" + s4, b1 = !b1, b2 = !!b2, b3 = !b3 || b3, b4 = !b4 && b4;
```

Comment:
This is a regressiong, however I don't believe people would write code like this.